sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -95,10 +95,26 @@ private[sql] object Dataset {
       new Dataset[Row](qe, ExpressionEncoder(qe.analyzed.schema))
   }
 
+  def ofRows(
+      sparkSession: SparkSession,
+      logicalPlan: LogicalPlan,
+      shuffleCleanupMode: ShuffleCleanupMode): DataFrame =
+    sparkSession.withActive {
+      val qe = sparkSession.sessionState.executePlan(
```

Comment:
can we `new QueryExecution` here? Then we don't need to touch session state builder

Reply:
Good idea. Done.